src/gui/sdlmain.cpp

```diff
@@ -1254,20 +1288,43 @@ SDL_Rect GFX_GetSDLSurfaceSubwindowDims(uint16_t width, uint16_t height)
 	return rect;
 }
 
+// kvas kompot pelmeni shashlyki kotlety bliny belyashi
+static void get_canvas_size(SCREEN_TYPES screen_type, int *w, int *h)
+{
+	switch (screen_type) {
+	case SCREEN_SURFACE: SDL_GetWindowSize(sdl.window, w, h); break;
+	case SCREEN_TEXTURE:
+		if (SDL_GetRendererOutputSize(sdl.renderer, w, h) != 0)
+			LOG_ERR("Failed to retrieve output size: %s",
+			        SDL_GetError());
+		break;
+#if C_OPENGL
+	case SCREEN_OPENGL: SDL_GL_GetDrawableSize(sdl.window, w, h); break;
+#endif
+	}
+}
+
 static SDL_Window *setup_window_pp(SCREEN_TYPES screen_type, bool resizable)
 {
 	assert(sdl.window);
 
-	int w, h;
-	if (sdl.desktop.fullscreen) {
-		SDL_GetWindowSize(sdl.window, &w, &h);
-	} else {
+	int w, h, canvas_width, canvas_height;
+
+	SDL_GetWindowSize(sdl.window, &w, &h);
+	assert(w > 0 && h > 0);
+
+	get_canvas_size(screen_type, &canvas_width, &canvas_height);
```

Comment:
Suggest width and height, and canvas width and height be packaged as an SDL_Point, similar to how `restrict_to_viewport_resolution()`, which return an SDL_Point type that holds width and height values. 

This will let you do something like:
```c++
const auto canvas_size = get_canvas_size(screen_type);
```

(Although, members of the point are .x and .y, which admittedly isn't as nice as .w and .h; I suppose and SDL_Rect could be used too)

Reply:
Makes sense. Not sure about window sizes, but done for output canvas.